### DISCUSSION TITLE
Dataframe commands documentation

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#7012] [Q&A] [answered]
Hi,
nu dataframe is really great!

I would like to learn to use the [commands](https://www.nushell.sh/book/dataframes.html#dataframe-commands), but I do not find commands documentation.

In example the `sample` command: what's the meaning of these flags?

```
-e, --replace - sample with replace
-u, --shuffle - shuffle sample
```

Thank you

**Comment by fdncred** [comment, marked as answer]
In order to use the dataframe command you need to install or build with `--features=extra`.

The `sample` command just creates sample data I think. It's not very intuitive. You can see on example with `help sample`.

**Reply by aborruso** [reply]
> In order to use the dataframe command you need to install or build with `--features=extra`.
> 
> The `sample` command just creates sample data I think. It's not very intuitive. You can see on example with `help sample`.

I'm using a dataframe and I have used nu help. But nu help gives me in example

```
-e, --replace - sample with replace
```

What's the meaning of it? Is it a global search and replace? Do I must insert after the flag the sting I search and the the string I want to use to replace it?

Thank you

**Reply by fdncred** [reply]
This is what I meant.
```
help sample
Create sample dataframe

Usage:
  > sample {flags}

Flags:
  -h, --help - Display the help message for this command
  -n, --n-rows <Int> - number of rows to be taken from dataframe
  -f, --fraction <Number> - fraction of dataframe to be taken
  -s, --seed <Number> - seed for the selection
  -e, --replace - sample with replace
  -u, --shuffle - shuffle sample

Examples:
  Sample rows from dataframe
  > [[a b]; [1 2] [3 4]] | into df | sample -n 1

  Shows sample row using fraction and replace
  > [[a b]; [1 2] [3 4] [5 6]] | into df | sample -f 0.5 -e
```
The last example demonstrates the `-e/-replace` flag.

**Reply by aborruso** [reply]
> The last example demonstrates the `-e/-replace` flag.

Ok, thank you. Running a lot of `[[a b]; [1 2] [3 4] [5 6]] | into df | sample -n 2 -e` I understood that the "replace" command, randomly swaps the values between all cells.
If that's correct, couldn't it be written somewhere? Can I do it? If so, where would you recommend me to do a PR?

**Reply by fdncred** [reply]
I'm not really positive what sample is supposed to do. I think it's based off this https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.sample.html

**Reply by aborruso** [reply]
> I'm not really positive what sample is supposed to do. I think it's based off this https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.sample.html

you are right `Allow values to be sampled more than once.`
